src/java.base/share/native/libjava/System.c

```diff
@@ -147,15 +147,7 @@ Java_jdk_internal_util_SystemProps_00024Raw_platformProperties(JNIEnv *env, jcla
     PUTPROP(propArray, _line_separator_NDX, sprops->line_separator);
 
     /* basic encoding properties, always non-NULL */
-#ifdef MACOSX
-    /*
-     * Since sun_jnu_encoding is now hard-coded to UTF-8 on Mac, we don't
-     * want to use it to overwrite native.encoding
-     */
     PUTPROP(propArray, _native_encoding_NDX, sprops->encoding);
-#else
-    PUTPROP(propArray, _native_encoding_NDX, sprops->sun_jnu_encoding);
-#endif
     PUTPROP(propArray, _sun_jnu_encoding_NDX, sprops->sun_jnu_encoding);
```

Comment:
Since we now got rid of platform depenence here, would it make sense to perform non-null assertion in Java level here? I remember in the last PR we added `assert` in platform C code.

Reply:
`Java level` -> `Platform independent C level`
Sorry

Reply:
Good idea. Done.